src/components/SubscribeButton/hooks/useSubscribeButton.ts

```diff
@@ -0,0 +1,126 @@
+import { useSession, useSupabaseClient } from '@supabase/auth-helpers-react'
+import { useWallet } from 'hooks/Wallet'
+import { useWalletSignIn } from 'hooks/WalletSignIn'
+import { ProjectNotification } from 'models/notifications/projectNotifications'
+import { useCallback, useEffect, useState } from 'react'
+import { Database } from 'types/database.types'
+import { emitErrorNotification } from 'utils/notifications'
+
+/**
+ * Hook to control the subscribe button.
+ * @param projectId The ID of the project to subscribe to.
+
+ * @returns The loading state, the subscribed state, and the subscribe function.
+ * @example const { loading, isSubscribed, subscribe } = useSubscribeButton({ projectId: 1 })
+ */
+export const useSubscribeButton = ({ projectId }: { projectId: number }) => {
+  const [loading, setLoading] = useState<boolean>(false)
+  const [isSubscribed, setIsSubscribed] = useState<boolean>(false)
+
+  const wallet = useWallet()
+  const signIn = useWalletSignIn()
+  const session = useSession()
+  const supabase = useSupabaseClient<Database>()
+
+  // If the user is not connected to a wallet, or the wallet is on an unsupported chain, don't show the button
+  const showSubscribeButton = wallet.isConnected && !wallet.chainUnsupported
+
+  const getUserNotificationsForProjectId = useCallback(
+    async ({ projectId, userId }: { projectId: number; userId: string }) => {
+      const { data, error } = await supabase
+        .from('user_subscriptions')
+        .select('notification_id')
+        .eq('user_id', userId)
+        .eq('project_id', projectId)
```

Comment:
woah. TIL. We can query supabase from the client? 🤔 . Do we want this?

Reply:
We can query from client ye - maybe, maybe not. We still have to authenticate the user, which poses a bit of a ux problem too. Not much we can do about it without having user sign on request though